pkg/backend/httpstate/client/client.go

```diff
@@ -50,6 +50,39 @@ const (
 	CopilotRequestTimeout = 20 * time.Second
 )
 
+// TemplatePublishOperationID uniquely identifies a template publish operation.
+type TemplatePublishOperationID string
+
+// StartTemplatePublishRequest is the request body for starting a template publish operation.
+type StartTemplatePublishRequest struct {
+	// Version is the semantic version of the template.
+	Version semver.Version `json:"version"`
+}
+
+// StartTemplatePublishResponse is the response from initiating a template publish.
+// It returns a presigned URL to upload the template archive.
+type StartTemplatePublishResponse struct {
+	// OperationID uniquely identifies the publishing operation.
+	OperationID TemplatePublishOperationID `json:"operationID"`
+	// UploadURLs contains the presigned URLs for uploading template artifacts.
+	UploadURLs TemplateUploadURLs `json:"uploadURLs"`
+}
+
+// TemplateUploadURLs contains the presigned URLs for uploading template artifacts.
+type TemplateUploadURLs struct {
+	// Archive is the URL for uploading the template archive.
+	Archive string `json:"archive"`
+}
+
+// PublishTemplateVersionCompleteRequest is the request body for completing a template publish operation.
+type PublishTemplateVersionCompleteRequest struct {
+	// OpID is the operation ID from the StartTemplatePublishResponse.
+	OpID TemplatePublishOperationID `json:"operationID"`
+}
+
+// PublishTemplateVersionCompleteResponse is the response from completing a template publish operation.
+type PublishTemplateVersionCompleteResponse struct{}
+
```

Comment:
This should be in `sdk/go/common/apitype`.

Reply:
That's where I had it originally but from our sync I got the impression that it's better to avoid updating SDK types until we stabilize these APIs and features.

What should I do? Add a comment mentioning that, or move things to `sdk/go` anyway?

Reply:
(Moving to `sdk/go` for now)